lib/rspec/core/configuration.rb

```diff
@@ -870,7 +874,7 @@ def full_description
       # Adds a formatter to the set RSpec will use for this run.
       #
       # @see RSpec::Core::Formatters::Protocol
-      def add_formatter(formatter, output=output_stream)
+      def add_formatter(formatter, output=output_wrapper)
```

Comment:
This is why it's necessary to wrap the passed in input here, a user can add their own output and we need to wrap it.